collect_app/src/main/java/org/odk/collect/android/preferences/dialogs/DeleteProjectDialog.kt

```diff
@@ -0,0 +1,204 @@
+                projectData.numberOfSentForms,
+                projectData.numberOfUnsentForms,
+                if (projectData.numberOfUnsentForms > 0) "⚠\uFE0F" else "",
+                projectData.numberOfDraftForms
+            )
+            binding.message.text = HtmlCompat.fromHtml(message, HtmlCompat.FROM_HTML_MODE_LEGACY)
+        }
+
+        viewModel.deleteProjectResult.observe(this) { result ->
+            dismiss()
+            when (result) {
+                is DeleteProjectResult.DeletedSuccessfullyCurrentProject -> {
+                    val newCurrentProject = result.newCurrentProject
+                    ActivityUtils.startActivityAndCloseAllOthers(
+                        requireActivity(),
+                        MainMenuActivity::class.java
+                    )
+                    ToastUtils.showLongToast(
+                        getString(
+                            org.odk.collect.strings.R.string.switched_project,
+                            newCurrentProject.name
+                        )
+                    )
+                }
+                is DeleteProjectResult.DeletedSuccessfullyLastProject -> {
+                    ActivityUtils.startActivityAndCloseAllOthers(
+                        requireActivity(),
+                        FirstLaunchActivity::class.java
+                    )
+                }
+                is DeleteProjectResult.DeletedSuccessfullyInactiveProject -> {
+                    // not possible here
+                }
+            }
+        }
+
+        return MaterialAlertDialogBuilder(requireContext())
+            .setView(binding.root)
+            .create()
+    }
+
+    class DeleteProjectViewModel(
+        private val projectDeleter: ProjectDeleter,
+        private val projectsDataService: ProjectsDataService,
+        private val formsRepositoryProvider: FormsRepositoryProvider,
+        private val instancesRepositoryProvider: InstancesRepositoryProvider,
+        scheduler: Scheduler
+    ) : ViewModel() {
+        private val trackableWorker = TrackableWorker(scheduler)
+
+        val isWorking = trackableWorker.isWorking
+
+        private val _projectData = MutableLiveData<ProjectData>()
+        val projectData: LiveData<ProjectData> = _projectData
+
+        private val _deleteProjectResult = MutableLiveData<DeleteProjectResult>()
+        val deleteProjectResult: LiveData<DeleteProjectResult> = _deleteProjectResult
+
+        private val project = projectsDataService.getCurrentProject().value!!
+
+        init {
+            trackableWorker.immediate {
+                val numberOfForms = formsRepositoryProvider.create(project.uuid).all.size
+                val instancesRepository = instancesRepositoryProvider.create(project.uuid)
+                val numberOfSentForms = instancesRepository.getCountByStatus(Instance.STATUS_SUBMITTED)
+                val numberOfUnsentForms = instancesRepository.getCountByStatus(
```

Comment:
Are these counts (other than the forms one) not available from `InstancesDataService`? Could we just use that and add a count to `FormsDataService`?